pkg/apis/core/v1beta1/helper/shoot.go

```diff
@@ -729,3 +730,26 @@ func IsOneWorkerPoolLowerKubernetes134(controlPlaneVersion *semver.Version, work
 	}
 	return atLeastOnePoolLowerKubernetes134, nil
 }
+
+// GetShootGardenerOperations returns the Shoot's gardener operations specified in the operation annotation.
+func GetShootGardenerOperations(annotations map[string]string) []string {
+	return splitAndTrimString(annotations[v1beta1constants.GardenerOperation], ";")
+}
+
+// GetShootMaintenanceOperations returns the Shoot's maintenance operations specified in the operation annotation.
+func GetShootMaintenanceOperations(annotations map[string]string) []string {
+	return splitAndTrimString(annotations[v1beta1constants.GardenerMaintenanceOperation], ";")
+}
+
+func splitAndTrimString(s, sep string) []string {
+	var res []string
+
+	if len(s) == 0 {
+		return res
+	}
+
+	for _, s0 := range strings.Split(s, sep) {
+		res = append(res, (strings.TrimSpace(s0)))
+	}
+	return res
```

Comment:
Nit, as I think it's a bit more clear when avoiding the `append`, but no strong preference:
```suggestion
	if len(s) == 0 {
		return nil
	}
    
    result := strings.Split(s, sep)
	for i := range result {
		result[i] = strings.TrimSpace(result[i])
	}
	
	return result
```